### ISSUE TITLE
Issue adding directory to PYTHONPATH Manager

### ISSUE THREAD

**Issue by obrienciaran** [spyder-ide/spyder#13537]
I am trying to add /usr/local/lib/python3.7/site-packages to the PYTHONPATH Manager. However, I receive an error informing me "This directory cannot be added to PATH. If you want to set a different Python interpreter, please go to Preferences > Main Interpreter".

However, I have already changed my interpreter to point to /usr/bin/python3

At the moment, I am using the rather annoying work around of putting the following at the top of all my code.

```
import sys
sys.path.append("/usr/local/lib/python3.7/site-packages")
```
Typing the following gives me the below. 
```
for p in sys.path: print(p)

/Users/user
/usr/local/lib/python3.7
/Users/user/opt/anaconda3/lib/python37.zip
/Users/user/opt/anaconda3/lib/python3.7
/Users/user/opt/anaconda3/lib/python3.7/lib-dynload

/Users/user/opt/anaconda3/lib/python3.7/site-packages
/Users/user/opt/anaconda3/lib/python3.7/site-packages/aeosa
/Users/user/opt/anaconda3/lib/python3.7/site-packages/IPython/extensions
/Users/user/.ipython
```

 I feel I am missing something obvious. Is there a reason why I cannot add this directory to my PATH?

**Comment by juanis2112** [comment]
Hi @obrienciaran. Thanks for reporting this. Did you restart your Spyder for the interpreter change to take effect?  
If you did, you are getting that message because Spyder doesn't allow to mix different versions of Python. Is that what you are trying to do? Are you trying to use the packages in your ``/usr/local`` directory along with the ones from Anaconda?

**Comment by obrienciaran** [comment]
Thanks for the response. Yes that is exactly what I am trying to do.

Ok I wasn't aware of that. The simple solution is to install packages via 'Conda' I guess?

**Comment by juanis2112** [comment]
Yes. That would be the best idea. Let me know if you need help with that.

**Comment by obrienciaran** [comment]
Thanks Juanis2112, this should be no problem. Thank you!

**Comment by hdino** [comment]
@juanis2112 I came across a similar situation when I tried to add the Python packages of [ROS 2 (Robot Operating System)](https://www.ros.org/) to PYTHONPATH. ROS' default installation path looks like this: `/opt/ros/foxy/lib/python3.8/site-packages`

Spyder's path manager will reject such paths, see [pathmanager.py:294-302](https://github.com/spyder-ide/spyder/blob/master/spyder/widgets/pathmanager.py#L294-L302). I would suggest replacing the error message in [line 359](https://github.com/spyder-ide/spyder/blob/master/spyder/widgets/pathmanager.py#L359) with a message box that informs the user that she/he might break some functionality and then decide whether to continue or not.

**Comment by ccordoba12** [comment]
@hdino, sorry but adding those packages would definitely break our console. That's why we decided to not allow it.

The right solution is to install spyder-kernels in your Python interpreter and make Spyder use it by going to

    Tools > Preferences > Main interpreter

**Comment by SiamakNV** [comment]
Hi, 

@ccordoba12: I am trying to add Gurobi to Spyder. I could do it in Anaconda. But cannot do it in Spyder (The anaconda version is slow so as you suggested I installed your version which is perfect). I used to add Gurobipy path to path manager in windows, but could not do it on mac and get the same error "This directory cannot be added to PATH". How can I possibly solve this problem? 

Thanks in advance, 
Siamak
